src/modules/rover_differential/RoverDifferential.cpp

```diff
@@ -0,0 +1,256 @@
+	if (_parameter_update_sub.updated()) {
+		parameter_update_s parameter_update;
+		_parameter_update_sub.copy(&parameter_update);
+		updateParams();
+	}
+
+	if (_vehicle_status_sub.updated()) {
+		vehicle_status_s vehicle_status{};
+		_vehicle_status_sub.copy(&vehicle_status);
+		_nav_state = vehicle_status.nav_state;
+	}
+
+	if (_vehicle_angular_velocity_sub.updated()) {
+		vehicle_angular_velocity_s vehicle_angular_velocity{};
+		_vehicle_angular_velocity_sub.copy(&vehicle_angular_velocity);
+		_vehicle_body_yaw_rate = vehicle_angular_velocity.xyz[2];
+	}
+
+	if (_vehicle_attitude_sub.updated()) {
+		vehicle_attitude_s vehicle_attitude{};
+		_vehicle_attitude_sub.copy(&vehicle_attitude);
+		_vehicle_attitude_quaternion = matrix::Quatf(vehicle_attitude.q);
+		_vehicle_yaw = matrix::Eulerf(_vehicle_attitude_quaternion).psi();
+	}
+
+	if (_vehicle_local_position_sub.updated()) {
+		vehicle_local_position_s vehicle_local_position{};
+		_vehicle_local_position_sub.copy(&vehicle_local_position);
+		Vector3f velocity_in_local_frame(vehicle_local_position.vx, vehicle_local_position.vy, vehicle_local_position.vz);
+		Vector3f velocity_in_body_frame = _vehicle_attitude_quaternion.rotateVectorInverse(velocity_in_local_frame);
+		_vehicle_forward_speed = velocity_in_body_frame(0);
+	}
+
+	// Navigation modes
+	switch (_nav_state) {
+	case vehicle_status_s::NAVIGATION_STATE_MANUAL: {
+			manual_control_setpoint_s manual_control_setpoint{};
+
+			if (_manual_control_setpoint_sub.update(&manual_control_setpoint)) {
+				_differential_setpoint.throttle = manual_control_setpoint.throttle;
+				_differential_setpoint.yaw_rate = manual_control_setpoint.roll * _param_rd_man_yaw_scale.get();
+
+			}
+
+			_differential_setpoint.closed_loop_yaw_rate = false;
+		} break;
+
+	case vehicle_status_s::NAVIGATION_STATE_ACRO: {
+			manual_control_setpoint_s manual_control_setpoint{};
+
+			if (_manual_control_setpoint_sub.update(&manual_control_setpoint)) {
+				_differential_setpoint.throttle = manual_control_setpoint.throttle;
+				_differential_setpoint.yaw_rate = math::interpolate<float>(manual_control_setpoint.roll,
+								  -1.f, 1.f,
+								  -_max_yaw_rate, _max_yaw_rate);
+			}
+
+			_differential_setpoint.closed_loop_yaw_rate = true;
+		} break;
+
+	case vehicle_status_s::NAVIGATION_STATE_AUTO_MISSION:
+	case vehicle_status_s::NAVIGATION_STATE_AUTO_RTL:
+		_differential_setpoint = _rover_differential_guidance.computeGuidance(_vehicle_yaw, _vehicle_forward_speed,
+					 _nav_state);
+		break;
+
+	default: // Unimplemented nav states will stop the rover
+		_differential_setpoint.throttle = 0.f;
+		_differential_setpoint.yaw_rate = 0.f;
+		_differential_setpoint.closed_loop_yaw_rate = false;
+		break;
+	}
+
+	float speed_diff_normalized = _differential_setpoint.yaw_rate;
+
+	// Closed loop yaw rate control
+	if (_differential_setpoint.closed_loop_yaw_rate) {
+		if (fabsf(_differential_setpoint.yaw_rate - _vehicle_body_yaw_rate) < 0.1f) {
```

Comment:
can you add comment for the 0.1 and/or make it a constant?

Reply:
implemented in 817bd21